walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,91 @@
 
 package com.google.sps;
 
+import com.google.sps.TimeRange;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Iterator;
+import java.lang.Math;
 
+/** Class that finds available slots to hold a meeting */
 public final class FindMeetingQuery {
+    
+/** Returns all the available time slots to schedule the meeting */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> availableSlots = new ArrayList<>(); 
+    List<TimeRange> occupiedSlots = new ArrayList<>();
+    Collection<String> requestAttendees = request.getAttendees();
+
+    // If duration time is longer than a day - no option
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return new ArrayList<TimeRange>();
+    }
+
+    // Add to occupiedSlots only event slots that include people who attend meeting request
+    for(Event event : events){
+        Set<String> eventAttendees = event.getAttendees();
+        if(!Collections.disjoint(eventAttendees, requestAttendees)){
+            occupiedSlots.add(event.getWhen());
+        }
+    }
+
+    // If none of the events have attendees that is in the request, all day is free
+    if(occupiedSlots.size() == 0){
+        availableSlots.add(TimeRange.WHOLE_DAY);
+        return availableSlots;
+    }
+
+    // Sort the events by start time
+    Collections.sort(occupiedSlots, TimeRange.ORDER_BY_START);
+
+    // If there is enough free time in the morning, add that slot
+    TimeRange currentEventTime = occupiedSlots.get(0);
+    int freeSlotStart = TimeRange.START_OF_DAY;
+    int freeSlotEnd = currentEventTime.start();
+    checkPlusAddSlot(availableSlots, request, freeSlotStart, freeSlotEnd, false);
+
+    // initialize the free slot candidate to be at the end of first event
+    freeSlotStart = currentEventTime.end();
+    int i = 0;
+    int j = 1;
```

Comment:
As far as I can tell 'j' is only used in the scope of the while loop. 

Consider initializing it there.